File: ToahaSBitwiseRule.java
/*Formula inventor: Hasin Israk Toaha (Constant T)-  SSC '25
 Java program wittten by: Hasin Israk Toaha
 Compiled in: Termux (Android 11)
 Compiler: javac (OpenJDK-17)

  Please give me credit if you use it anywhere
  Contact: toaha.banaripara@gmail.com

*/

import java.math.*;
import java.util.*;

public class ToahaSBitwiseRule
{
    public static boolean isPrime(BigDecimal num) 
    {
        if (num.compareTo(BigDecimal.ONE) <= 0) return false;
        BigDecimal two = BigDecimal.valueOf(2);
        for (BigDecimal i = two; i.multiply(i).compareTo(num) <= 0; i = i.add(BigDecimal.ONE)) 
        {
            if (num.remainder(i).compareTo(BigDecimal.ZERO) == 0) return false;
        }
        return true;
    }

    public static void main(String[] args)
    {
        System.out.print("Enter the terms: ");
        Scanner sc = new Scanner(System.in);
        BigDecimal terms = sc.nextBigDecimal();

        long startTime = System.currentTimeMillis();

        for (BigDecimal i = BigDecimal.ONE; i.compareTo(terms) <= 0; i = i.add(BigDecimal.ONE)) 
        {
            if (i.remainder(BigDecimal.valueOf(2)).compareTo(BigDecimal.ZERO) == 0) 
            {
                BigDecimal next = i.add(BigDecimal.ONE); 
                if (isPrime(next) || next.remainder(BigDecimal.valueOf(3)).compareTo(BigDecimal.ZERO) == 0) 
                {
                    BigDecimal andResult = new BigDecimal(i.toBigInteger().and(next.toBigInteger()));
                    BigDecimal orResult = new BigDecimal(i.toBigInteger().or(next.toBigInteger())); 
                    System.out.printf("%s & %s = %s%n%s | %s = %s%n%n", i, next, andResult, i, next, orResult);
                }
            }
        }

        long endTime = System.currentTimeMillis();

        System.out.printf("Time taken: %.8f Seconds%n",(endTime - startTime) / 1000.0);
        sc.close();
    }
}

File: ToahaSBitwiseRule.py

#Format inventor: Hasin Israk Toaha (Constant T)

#Python code written by: Hasin Israk Toaha

#Executed in: Termux (Android 11)


from decimal import *
from time import *

class ToahaSBitwiseRule:
    def __init__(self, terms):
        self.terms = Decimal(terms)

    def is_prime(self, num):
        if num <= 1:
            return False
        for i in range(2, int(num.sqrt()) + 1):
            if num % i == 0:
                return False
        return True

    def calculate_bitwise_operations(self):
        start_time = time()

        for i in range(1, int(self.terms) + 1):
            if i % 2 == 0:
                next_num = i + 1
                if self.is_prime(Decimal(next_num)) or next_num % 3 == 0:
                    and_result = i & next_num
                    or_result = i | next_num
                    print(f"{i} & {next_num} = {and_result}")
                    print(f"{i} | {next_num} = {or_result}\n")

        end_time = time()
        print(f"Time taken: {end_time - start_time:.8f} Seconds")

def main():
    terms = input("Enter the terms: ")
    bitwise_rule = ToahaSBitwiseRule(terms)
    bitwise_rule.calculate_bitwise_operations()

if __name__ == "__main__":
    main()